routers/web/repo/helper.go

```diff
@@ -20,3 +23,22 @@ func MakeSelfOnTop(doer *user.User, users []*user.User) []*user.User {
 	}
 	return users
 }
+
+func HandleGitError(ctx *context.Context, msg string, err error) {
+	if git.IsErrNotExist(err) {
+		refType := ""
+		switch {
+		case ctx.Repo.IsViewBranch:
+			refType = "branch"
+		case ctx.Repo.IsViewTag:
+			refType = "tag"
+		case ctx.Repo.IsViewCommit:
+			refType = "commit"
+		}
+		ctx.Data["NotFoundPrompt"] = ctx.Locale.Tr("repo.tree_path_not_found", ctx.Repo.TreePath, refType, url.PathEscape(ctx.Repo.RefName))
```

Comment:
I recommend a slightly different approach:
```suggestion
		ctx.Data["NotFoundPrompt"] = ctx.Locale.Tr("repo.tree_path_not_found_" + refType, ctx.Repo.TreePath, url.PathEscape(ctx.Repo.RefName))
```
Otherwise we won't be able to localize `branch`/`tag`/`commit`